modules/core/04-channel/types/msgs.go

```diff
@@ -566,33 +563,35 @@ func (msg MsgChannelUpgradeTry) ValidateBasic() error {
 	if err := host.PortIdentifierValidator(msg.PortId); err != nil {
 		return errorsmod.Wrap(err, "invalid port ID")
 	}
+
 	if !IsValidChannelID(msg.ChannelId) {
 		return ErrInvalidChannelIdentifier
 	}
-	if msg.CounterpartyChannel.State != INITUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.CounterpartyChannel.State)
-	}
-	if msg.CounterpartySequence == 0 {
-		return errorsmod.Wrap(ibcerrors.ErrInvalidSequence, "counterparty sequence cannot be 0")
+
+	if len(msg.ProposedUpgradeConnectionHops) == 0 {
+		return errorsmod.Wrap(ErrInvalidUpgrade, "proposed connection hops cannot be empty")
 	}
-	if msg.ProposedUpgradeChannel.State != TRYUPGRADE {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", TRYUPGRADE, msg.CounterpartyChannel.State)
+
+	if msg.UpgradeTimeout.Height.IsZero() && msg.UpgradeTimeout.Timestamp == 0 {
+		return errorsmod.Wrap(ErrInvalidUpgradeTimeout, "timeout height or timeout timestamp must be non-zero")
 	}
-	if msg.TimeoutHeight.IsZero() {
-		return errorsmod.Wrap(ibcerrors.ErrInvalidHeight, "timeout height must be non-zero")
+
+	if err := msg.CounterpartyProposedUpgrade.Fields.ValidateBasic(); err != nil {
+		return errorsmod.Wrap(err, "error validating counterparty upgrade fields")
 	}
-	if msg.TimeoutTimestamp == 0 && msg.TimeoutHeight.IsZero() {
-		return errorsmod.Wrap(ErrInvalidUpgradeTimeout, "invalid upgrade timeout timestamp or timeout height")
+
+	if msg.CounterpartyUpgradeSequence == 0 {
+		return errorsmod.Wrap(ibcerrors.ErrInvalidSequence, "counterparty sequence cannot be 0")
```

Comment:
I know this was here before but I feel like `ibcerrors.ErrInvalidSequence` shouldn't refer to the upgrade. I think the base error of `ErrInvalidUpgrade` makes sense. I'd also be okay with a dedicated error for the upgrade sequence

```suggestion
		return errorsmod.Wrap(ErrInvalidUpgrade, "counterparty upgrade sequence cannot be 0")
```